### PR TITLE
close #709 - check existence of radiation restart file

### LINKED ISSUE #709: radiation restart crashes if start time is not reached yet
The new *libSplash* based restart of the **Radiation plugin** fails with
```
[1,0]<stderr>:terminate called after throwing an instance of 'splash::DCException'
[1,0]<stderr>:  what():  Exception for SerialDataCollector::openRead: File not found (checkpoints/radRestart_4000_0_0_0.h5)
```
when the start time of the radiation plugin ` --radiation_e.start 4300` is not yet reached.

### PR DESCRIPTION
**Pull request by PrometheusPi**
This pull request adds a check to the **Radiation Plugin** testing if the restart file exists and tus closes #709 (partially).

- [x] verify that missing restart files still result in running simulation

### CODE REVIEW

File: src/picongpu/include/plugins/radiation/Radiation.hpp [hunk 1]
```diff
@@ -595,36 +597,44 @@ class Radiation : public ISimulationPlugin
       /* add to standard ending added by libSpash for SerialDataCollector */
       filename << name << timeStep << "_0_0_0.h5";
 
-      HDF5dataFile.open(filename.str().c_str(), fAttr);
-
-      typename PICToSplash<double>::type radSplashType;
+      /* check if restart file exists */
+      if( !boost::filesystem::exists(filename.str()) )
+      {
+        log<radLog::SIMULATION_STATE > ("radiation restart file not found (%1%) - start with zero values") % filename.str();
```
Comment (ax3l) [inline src/picongpu/include/plugins/radiation/Radiation.hpp]: usually we use `INPUT_OUTPUT` for restart related logs

**Comment by ax3l** [comment]
pls try to use consistently `Radiation:` or `[Radiation]` as a prefix for your log/cout messages.

**Comment by ax3l** [comment]
afterwards, if you squash all commits to one, I can backport it so it goes in the `0.1` release.

**Comment by PrometheusPi** [comment]
@ax3l are you shure we should squash this commit to one?
(It changes IO log levels, IO behaviour and file checks.)

**Comment by ax3l** [comment]
the last changes are newer then my message.

nevertheless: if you want it in `0.1` - squash it to one commit. :shipit: 
if you don't care about the upcoming release: use at maximum 2 commits: one for the log and one for the algorithm change. that's more than enough.

**Comment by ax3l** [comment]
update: I did not say "please add just an other log level" - I meant "use the default PIConGPU `INPUT_OUTPUT` log level" - in your case please do *not* blow up the extra log just by one more level.

rather unify it to use the PIConGPU levels and to reduce the code base.

**Comment by PrometheusPi** [comment]
too late - I just rebased

**Comment by ax3l** [comment]
just do not push it.

File: src/picongpu/include/plugins/radiation/debug/PIConGPUVerboseLogRadiation.hpp [hunk 2]
```diff
@@ -38,6 +38,7 @@ DEFINE_VERBOSE_CLASS(PIConGPUVerboseRadiation)
     DEFINE_LOGLVL(2,SIMULATION_STATE);
```
Comment (ax3l) [inline src/picongpu/include/plugins/radiation/debug/PIConGPUVerboseLogRadiation.hpp]: yes, there is absolutely no reason to use just an other log level here.

we should refactor this in a follow up pull request (post `0.1` release)

**Comment by ax3l** [comment]
well done! get's backported! :+1:

**Pull request merged**